core/src/main/java/com/opensymphony/xwork2/interceptor/ChainingInterceptor.java

```diff
@@ -160,7 +160,20 @@ private void copyStack(ActionInvocation invocation, CompoundRoot root) {
         Map<String, Object> ctxMap = invocation.getInvocationContext().getContextMap();
         for (Object object : list) {
             if (shouldCopy(object)) {
-                reflectionProvider.copy(object, invocation.getAction(), ctxMap, prepareExcludes(), includes);
+                String actionConfigClassName = invocation.getProxy().getConfig().getClassName();
+                Object action = invocation.getAction();
+                if (action.getClass().getName().equals(actionConfigClassName))
+                    reflectionProvider.copy(object, action, ctxMap, prepareExcludes(), includes, null);
+                else {	//only setting properties defined in the given config class e.g. to skip proxy properties (WW-4105)
+                    Class <?> editable;
+                    try {
+                        editable =  Class.forName(actionConfigClassName);
+                    } catch (ClassNotFoundException e) {
+                        LOG.warn("An unexpected state. This may be due to a bug. Please report this: ClassNotFoundException" + actionConfigClassName);
```

Comment:
I also fixed Indentations to use 4 spaces instead of tab, both in my current PRs and my eclipse. Sorry, I'll be more careful about formatting in my next works